### PR TITLE
add blacklist files to validation set and re-train with base lr 0.4

### PR DESCRIPTION
**Pull request by jph00**
The last JSON file I provided incorrectly stated our base LR was 0.4 - it was actually 0.3. I just re-ran with the blacklist added back in to the validation set, this time actually using an LR of 0.4. No code was changed - the LR change is a hyperparam that was already in the JSON.

### CODE REVIEW

File: ImageNet/train/fastai_pytorch.tsv [hunk 1]
```diff
@@ -1,46 +1,46 @@
...
+11	0.39372730833333336	43.904	69.584
+12	0.42229111444444445	38.816	65.712
+13	0.45084065944444446	39.520	66.960
+14	0.4794358952777778	42.640	70.208
+15	0.5098701902777778	42.448	68.416
+16	0.5416913427777778	40.160	67.760
+17	0.5703782808333334	44.192	70.448
+18	0.6707612511111111	51.856	77.888
+19	0.7648842469444445	49.728	75.264
+20	0.858128875	42.592	69.136
```
Comment (frankchn) [inline ImageNet/train/fastai_pytorch.tsv:22]: Hi @jph00, just curious, do you know why the first 20 epochs (150+ seconds per epoch) are twice as fast as the next 20 (340+ seconds per epoch)?

Is it something pytorch specific or would it be something you are doing different apart from learning rate changes?

**Comment by jph00** [comment]
We use a progressive resizing technique, somewhat like Progressive GAN
and EDSR super-resolution papers. It's introduced discussed in
course.fast.ai lessons 1 & 2, and we'll be writing up the approach in
more detail after this competition.
--
Jeremy Howard


On Thu, Apr 26, 2018, at 11:04 AM, Frank Chen wrote:
> *@frankchn* commented on this pull request.


> 
> In ImageNet/train/fastai_pytorch.tsv[1]:


> > +5 0.2013068286111111 32.592 59.008 +6 0.23029296222222223 40.080
> > 67.408 +7 0.2658237766666667 38.224 65.184 +8 0.29882442277777777
> > 41.344 68.432 +9 0.3276660186111111 40.704 67.072 +10
> > 0.3562790472222222 44.096 70.336 +11 0.39372730833333336 43.904
> > 69.584 +12 0.42229111444444445 38.816 65.712 +13 0.45084065944444446
> > 39.520 66.960 +14 0.4794358952777778 42.640 70.208 +15
> > 0.5098701902777778 42.448 68.416 +16 0.5416913427777778 40.160
> > 67.760 +17 0.5703782808333334 44.192 70.448 +18 0.6707612511111111
> > 51.856 77.888 +19 0.7648842469444445 49.728 75.264 +20 0.858128875
> > 42.592 69.136
>> Hi @jph00[2], just curious, do you know why the first 20 epochs
> (150+ seconds per epoch) are twice as fast as the next 20 (340+
> seconds per epoch)?> Is it something pytorch specific or would it be something you are
> doing different apart from learning rate changes?> — You are receiving this because you were mentioned. Reply to this
> email directly, view it on GitHub[3], or mute the thread[4].> 


Links:

  1. https://github.com/stanford-futuredata/dawn-bench-entries/pull/42#discussion_r184481645
  2. https://github.com/jph00
  3. https://github.com/stanford-futuredata/dawn-bench-entries/pull/42#pullrequestreview-115688265

**Comment by frankchn** [comment]
Cool, thanks for the references!

**Comment by codyaustun** [comment]
Looks good to me. Thanks for updating.

File: ImageNet/train/fastai_pytorch.json [hunk 2]
```diff
@@ -3,11 +3,12 @@
     "author": "fast.ai + students team: Jeremy Howard, Andrew Shaw, Brett Koonce, Sylvain Gugger",
     "authorEmail": "j@fast.ai",
     "framework": "fastai / pytorch",
-    "codeURL": "https://github.com/fastai/imagenet-fast/blob/e47cc1a0a5e1b14c75b26a3ed67a4346124568a7/aws/do_nosh.sh",
+    "codeURL": "https://github.com/fastai/imagenet-fast/blob/c4b225555e333a1a2702d2b291b5082bfa6d6a0a/aws/do_nosh.sh",
```
Comment (tim951009) [inline ImageNet/train/fastai_pytorch.json:6]: It seemed the validation set still excluded blacklisted files.

do_nosh.sh called upload_scripts/train2.sh, which still called upload_scripts/blacklist.sh

https://github.com/fastai/imagenet-fast/blob/c4b225555e333a1a2702d2b291b5082bfa6d6a0a/aws/upload_scripts/train2.sh#L72

https://github.com/fastai/imagenet-fast/blob/c4b225555e333a1a2702d2b291b5082bfa6d6a0a/aws/upload_scripts/blacklist.sh

**Comment by jgong5** [comment]
There are actually lots of changes between this new submission and the last one if you compare the two commit ids. They also seem to change the model in-between. I think changing any code (including hyper-parameters and models etc.) for a re-submission SHOULD NOT be allowed after the competition deadline. Suggest to rollback this PR.
```
diff --git a/imagenet_nv/models/resnet.py b/imagenet_nv/models/resnet.py
index d2c6bcb..254e542 100644
--- a/imagenet_nv/models/resnet.py
+++ b/imagenet_nv/models/resnet.py
@@ -1,6 +1,7 @@
import torch.nn as nn
import math
import torch.utils.model_zoo as model_zoo
+from .layers import *


model_urls = {
@@ -17,6 +18,12 @@ def conv3x3(in_planes, out_planes, stride=1):
     return nn.Conv2d(in_planes, out_planes, kernel_size=3, stride=stride,
                      padding=1, bias=False)

+def bn1(planes):
+    m = nn.BatchNorm1d(planes)
+    m.weight.data.fill_(1)
+    m.bias.data.zero_()
+    return m
+
def bn(planes, init_zero=False):
     m = nn.BatchNorm2d(planes)
     m.weight.data.fill_(0 if init_zero else 1)
@@ -39,19 +46,17 @@ class BasicBlock(nn.Module):

     def forward(self, x):
         residual = x
+        if self.downsample is not None: residual = self.downsample(x)

         out = self.conv1(x)
-        out = self.bn1(out)
         out = self.relu(out)
+        out = self.bn1(out)

         out = self.conv2(out)
-        out = self.bn2(out)
-
-        if self.downsample is not None:
-            residual = self.downsample(x)

         out += residual
         out = self.relu(out)
+        out = self.bn2(out)

         return out
```
